src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -130,11 +225,12 @@ bool EffectDummyCreature_lazy_peon_awake(Unit* pCaster, uint32 uiSpellId, SpellE
     // always check spellid and effectindex
     if (uiSpellId == SPELL_AWAKEN_PEON && uiEffIndex == EFFECT_INDEX_0)
     {
+        //TODO some of these checks are now redundant, should be removed by someone who knows how to do so without breaking things
```

Comment:
I'm not doubting myself on the things you brought up, I'm doubting myself on the true/false as I don't really understand what true/false do in that context exactly. If you could clear that up I could easily clean up that check thing.